### PR TITLE
ftp: Add rust parsers for active mode port handling 

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #4151

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [2916](https://redmine.openinfosecfoundation.org/issues/2916)

Describe changes:
- Addresses review comments
- Removes trailing NULL handling

Thanks @jasonish!

### CODE REVIEW

File: src/app-layer-ftp.c [hunk 1]
```diff
@@ -663,7 +600,7 @@ static int FTPParseRequest(Flow *f, void *ftp_state,
                 if (state->current_line_len + 1 > state->port_line_size) {
                     /* Allocate an extra byte for a NULL terminator */
                     ptmp = FTPRealloc(state->port_line, state->port_line_size,
-                                      state->current_line_len + 1);
+                                      state->current_line_len );
                     if (ptmp == NULL) {
                         if (state->port_line) {
                             FTPFree(state->port_line, state->port_line_size);
```
Comment (victorjulien) [inline src/app-layer-ftp.c]: for some reason this came back:
```
  CC       app-layer-ftp.o
app-layer-ftp.c:606:29: warning: Use of memory after it is freed
                            FTPFree(state->port_line, state->port_line_size);
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

Comment (jlucovsky) [reply src/app-layer-ftp.c]: I wasn't able to reproduce this warning with clang-6 nor clang-8. What version was used when the warning was generated?

File: rust/src/ftp/mod.rs [hunk 2]
```diff
@@ -56,6 +78,24 @@ named!(pub ftp_pasv_response<u16>,
 );
 
 
+#[no_mangle]
+pub extern "C" fn rs_ftp_active_port(input: *const u8, len: u32) -> u16 {
+    let buf = CompleteByteSlice(build_slice!(input, len as usize));
+    match ftp_active_port(buf) {
+        Ok((_, dport)) => {
+            return dport;
+        }
```
Comment (shivan1b) [inline rust/src/ftp/mod.rs]: Is  a comma not required after this closing brace?

File: src/app-layer-ftp.c (hunk 1 above)
Comment (victorjulien) [reply src/app-layer-ftp.c]: This is with:
```
$ CC=clang-8 scan-build-8 make
```
We don't run it in the CI

**Comment by jlucovsky** [comment]
Continued in #4212

**Pull request closed without merging**